### DISCUSSION TITLE
add makeTag and downsample to public API

### DISCUSSION THREAD

**Discussion by robstoll** [ajalt/mordant#197] [Ideas]
I consider using mordant where currently my use case is basically rendering text + style.
Currently, I have:
```
fun Terminal.render(text: String, style: TextStyle): String =
  t.render(Lines(listOf(Line(listOf(Span.word(text, red))))))
```
which works but has unnecessary overhead. How about making downsample and makeTag public?
Then I could have my:
```
fun Terminal.render(text: String, style: TextStyle) : String {
    val newStyle = downsample(style, info.ansiLevel, info.ansiHyperLinks)
    return makeTag(DEFAULT_STYLE, newStyle)
}

**Comment by ajalt** [comment]
There's already a `render` overload that takes a string, so you can do

```kotlin
terminal.render(style(string))
```

**Reply by robstoll** [reply]
On top of my head, that's the overload which parses the string, right? If so, then I guess that's even more overhead than my current solution.

**Reply by ajalt** [reply]
String parsing is just a regex, it should be more than fast enough to call frequently.
